### ISSUE TITLE
Single-Run Data for Type-C and D Models

### ISSUE THREAD

**Issue by alicewuhhh** [cvnlab/GLMsingle#161]
I have a question about running type-C and D models on single-run data. In our study, we have two runs of a motor task, with each run consisting of 10 identical repeated blocks. We want to analyze each run separately. I attempted to apply the type-C model with the GLMdenoise option, but it seems that this option requires more than one run. According to the documentation, if I set the pcstop option to -B, GLMdenoise will run, but cross-validation won't be performed for the type-C model.

So, my questions are:
1) If there is only one run, how can I use the GLMsingle type-C and D models to denoise the data (obtaining GLM nuisance regressors)?
2) If I combine the two runs, can I still analyze them separately considering each block will produce a single beta coefficient?

Thank you!

**Comment by kendrickkay** [comment]
> I have a question about running type-C and D models on single-run data. In our study, we have two runs of a motor task, with each run consisting of 10 identical repeated blocks. We want to analyze each run separately. I attempted to apply the type-C model with the GLMdenoise option, but it seems that this option requires more than one run. According to the documentation, if I set the pcstop option to -B, GLMdenoise will run, but cross-validation won't be performed for the type-C model.
> 

Right.  You need to give GLMsingle two runs in order to do the (default) cross-validation procedure for type C.  However, if you use the "-B" option, you can circumvent the cross-validation and still perform the GLMdenoise component of GLMsingle.


> So, my questions are:
> 
> If there is only one run, how can I use the GLMsingle type-C and D models to denoise the data (obtaining GLM nuisance regressors)?
If you supply GLMsingle with only one run, you can't really use the typeC and typeD models.  That is, unless you circumvent it using the -B option (as described above) and/or using the "single scalar value" case for the <fracs> option.  Of course, the tricky issue is how to know what exact parameter to use for these two options.


> If I combine the two runs, can I still analyze them separately considering each block will produce a single beta coefficient?

If you give two runs to GLMsingle, it will, ultimately, supply you a single beta coefficient for each block.  If i understand correctly, you have multiple blocks in each of your two runs.  So it seems like you should be fine, since you can decide how to split/combine/use the block-wise betas that you get from the data from the two runs.
